src/rtt.c

```diff
@@ -223,7 +223,7 @@ static void find_rtt(target_s *const cur_target)
 static rtt_retval_e read_rtt(target_s *const cur_target, const uint32_t i)
 {
 	/* copy data from recv_buf to target rtt 'down' buffer */
-	if (rtt_nodata())
+	if (rtt_nodata(i - rtt_num_up_chan))
```

Comment:
A note in the code about why `i - rtt_num_up_chan` is correct and what that expression really means would be super helpful please.